Map feature

cynthia-doc opened the pull request:
Map with clickable markers showing info of photoed locations in the travel corner section.

preview: https://htmlpreview.github.io/?https://github.com/cynthia-doc/my-portfolio/blob/mapFeature/portfolio/src/main/webapp/index.html
(the preview does not load the map so please refer to the deployed website)

deployed website: https://hcynth-step-2020.wl.r.appspot.com/

cynthia-doc committed:
map with markers and info to photoed locations

ricazhang commented on portfolio/src/main/webapp/index.html:
```diff
@@ -5,7 +5,8 @@
     <link rel="stylesheet" href="style.css">
-    <script src="script.js"></script>
+    <script src="https://maps.googleapis.com/maps/api/js?key=AIzaSyCfRqi1JtYmD4x7zPtLW2NXE9TCVrMMirM&callback=initMap"></script>
+    
```
extra newline?

ricazhang commented on portfolio/src/main/webapp/index.html:
```diff
@@ -36,7 +37,7 @@ <h2>About</h2>
             
         </div>
-        <div id="movie-container"><body onload="onLoad()" id="movies" class="section"></div><br>
+        <div id="movie-container"><body id="movies" class="section"></div><br>
```
was it intentional to add a body tag here?

cynthia-doc replied:
This section wasn't showing up properly before so I added the body tag here intentionally, but later found out that the problem was something else.

ricazhang commented on portfolio/src/main/webapp/index.html:
```diff
@@ -175,7 +176,10 @@ <h2>Travel Log</h2>
                 <a class="prev" onclick="plusSlides(-1)">&#10094;</a>
                 <a class="next" onclick="plusSlides(1)">&#10095;</a>
             </div>
+            
```
exrta?

ricazhang commented on portfolio/src/main/webapp/script.js:
```diff
@@ -92,3 +91,200 @@ function deleteComments() {
+    });
+
+    /** Sapporo */
+    const spLatlng = {lat: 43.0618, lng: 141.3545};
+
+    const spMarker = new google.maps.Marker({
+        position: spLatlng,
+        map: map,
+        title: "Sapporo"
+    });
+
+    const spStr = '<p><b>Sapporo</b>, capital of the mountainous northern Japanese island of Hokkaido, is famous for its beer, ' + 
+        'skiing and annual Sapporo Snow Festival featuring enormous ice sculptures. It is the fifth largest city in Japan. ' + 
+        'Ski hills and jumps from the 1972 Winter Olympics are scattered within the city limits, and the Sapporo Beer Museum traces the city’s brewing history.</p>';
+
+    const spInfo = new google.maps.InfoWindow({content: spStr});
+
+    spMarker.addListener('click', function() {
+        spInfo.open(map, spMarker);
+    });
+
+    /** Hangzhou */
+    const hzLatlng = {lat: 30.2741, lng: 120.1551};
+
+    const hzMarker = new google.maps.Marker({
+        position: hzLatlng,
+        map: map,
+        title: "Hangzhou"
+    });
+    
+    const hzStr = '<p><b>Hangzhou</b>, the capital of China’s Zhejiang province, is the southern terminus of the ancient ' + 
+        'Grand Canal waterway, which originates in Beijing. It is known for its West Lake, which is celebrated by poets and artists since the 9th century ' + 
+        'and encompasses islands, temples, pavilions, gardens and arched bridges.</p>';
+
+    const hzInfo = new google.maps.InfoWindow({content: hzStr});
+
+    hzMarker.addListener('click', function() {
+        hzInfo.open(map, hzMarker);
+    });
+}
```
it would be better to create a json array/object or json file that stores all of this information. Then you can loop through the object and create InfoWindows and add listeners for each location.

cynthia-doc replied:
fixed it on the Blobstore branch (which will be merged to this branch:))

ricazhang commented on portfolio/src/main/webapp/style.css:
```diff
@@ -225,6 +225,11 @@ label[for="max-display"] {
+#map {
+    height: 400px; 
+    width: 50%;
+    margin: 0 auto 20px auto;
```
please use the shorthand!

ricazhang commented on portfolio/src/main/webapp/script.js:
```diff
@@ -36,10 +36,10 @@ function showSlides(n) {
-    movieRecommender();
-    getComments();
-}
+document.getElementById('movie-container').addEventListener("load", movieRecommender());
+window.addEventListener("load", getComments());
+document.getElementById('map').addEventListener("load", createMap());
```
Single quote strings https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes

Alfredhgoog commented on portfolio/src/main/webapp/script.js:
```diff
@@ -36,10 +36,10 @@ function showSlides(n) {
-    movieRecommender();
-    getComments();
-}
+document.getElementById('movie-container').addEventListener("load", movieRecommender());
+window.addEventListener("load", getComments());
+document.getElementById('map').addEventListener("load", createMap());
```
I don't think you have to add brackets for these event listener registrations. You want to register with the function object, not the result of the function invocation. By adding the brackets, you're calling the function and executing it immediately even before the element is loaded.

cynthia-doc replied:
Just double checking, for the brackets, are you referring to the brackets after function calls? For example, I should have addEventListener("load", getComments) instead of addEventListener("load", getComments() )?

Alfredhgoog replied:
Yes that's what I meant.

cynthia-doc committed:
servlets for blobstore upload

cynthia-doc committed:
added a section in the Comment area to upload movie poster

cynthia-doc committed:
shorthand for #map margin

cynthia-doc committed:
Update script.js

Switch double quotes to singles, removing () after function names when adding event listener

cynthia-doc committed:
Update index.html

removed extra new lines

cynthia-doc committed:
fixed style issues, added function documentation, modified variable names to more appropriate ones

cynthia-doc committed:
storing location markers and info string in two arrays to loop through when creating markers on map

cynthia-doc committed:
removed debugging lines

cynthia-doc committed:
Merge branch 'mapFeature' into Blobstore

cynthia-doc committed:
Merge pull request #7 from cynthia-doc/Blobstore

Blobstore to upload movie poster
